### PR TITLE
bpo-37915: Fix comparison between tzinfo objects and timezone objects

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue37915

Automerge-Triggered-By: @pablogsal

### CODE REVIEW

File: Lib/test/datetimetester.py [hunk 1]
```diff
@@ -413,6 +413,9 @@ def test_offset_boundaries(self):
                 with self.assertRaises(ValueError):
                     timezone(delta)
 
+    def test_comparison_with_tzinfo(self):
+        self.assertNotEqual(timezone.utc, tzinfo())
```
Comment (pganssle) [inline Lib/test/datetimetester.py:419]: I was going to say harumph about constructing a `tzinfo` object, but this is apparently done elsewhere in the tests, with this comment:

```
        # In order to allow subclasses to get pickled, the C implementation
        # wasn't able to get away with having __init__ raise
        # NotImplementedError
```

In any case, I think maybe a comment referencing `bpo-37915` would be a good idea.

File: Misc/NEWS.d/next/Library/2019-08-22-16-13-27.bpo-37915.xyoZI5.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Fix a segmentation fault that appear when comparing instances of
```
Comment (pganssle) [inline Misc/NEWS.d/next/Library/2019-08-22-16-13-27.bpo-37915.xyoZI5.rst]:
```suggestion
Fix a segmentation fault that appeared when comparing instances of
```

File: Modules/_datetimemodule.c [hunk 3]
```diff
@@ -32,6 +32,8 @@
 #define PyTZInfo_Check(op) PyObject_TypeCheck(op, &PyDateTime_TZInfoType)
 #define PyTZInfo_CheckExact(op) (Py_TYPE(op) == &PyDateTime_TZInfoType)
 
+#define PyTimezone_Check(op) PyObject_TypeCheck(op, &PyDateTime_TimeZoneType)
```
Comment (pganssle) [inline Modules/_datetimemodule.c:35]: I believe that not having a `PyTimezone_Check` was a deliberate omission by @abalkin and that `timezone` is not intended to be subclassed.

I guess this isn't public anyway, but considering that it's not, I think there's no need to define both `PyTimezone_Check` and `PyTimezone_CheckExact`, just define whichever one you need. I'm thinking `PyTimezone_CheckExact` is most in keeping with the "`timezone` shouldn't be subclassed" philosophy (we can always add subclass support later, after all, while taking it away would be a PITA).

**Review by pganssle — CHANGES_REQUESTED** [review]

File: Modules/_datetimemodule.c (hunk 3 above)
Comment (pablogsal) [reply Modules/_datetimemodule.c:35]: ~`PyTZInfo_Check` was used elsewhere and I prefer that this PR does not touch those functions. I can inline the macro if you prefer~

Sorry, I totally misread the comment. I have removed the other macro.

**Comment by pablogsal** [comment]
I have made the requested changes; please review again

**Comment by tirkarthi** [comment]
3.7 also has this change with https://github.com/python/cpython/commit/6ed20e54e4c110e9adcfb70aba85310625e3edb4 and would require a backport. Thanks Pablo for the fix.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**